imbi/clients/aws.py

```diff
@@ -0,0 +1,41 @@
+import boto3
+
+
+def get_credentials(role_arn: str, username: str, id_token: str):
+    """Fetch temporary AWS credentials from STS."""
+    sts_client = boto3.client('sts')
+    return sts_client.assume_role_with_web_identity(
+        RoleArn=role_arn,
+        RoleSessionName=f'{username}-imbi',
+        WebIdentityToken=id_token)
+
+
+def get_parameters_by_path(
+        path: str,
+        access_key_id: str,
+        secret_access_key: str,
+        session_token: str) -> list:
+    """Fetch all SSM parameters under the specified path."""
+    ssm_client = boto3.client(
+        'ssm',
+        aws_access_key_id=access_key_id,
+        aws_secret_access_key=secret_access_key,
+        aws_session_token=session_token)
+
+    params = []
+    params_response = ssm_client.get_parameters_by_path(
+        Path=path,
+        Recursive=True,
+        WithDecryption=True,
+        MaxResults=10)
+    while True:
+        params.extend(params_response['Parameters'])
+        if 'NextToken' not in params_response:
+            break
+        params_response = ssm_client.get_parameters_by_path(
+            Path=path,
+            Recursive=True,
+            MaxResults=10,
+            WithDecryption=True,
+            NextToken=params_response['NextToken'])
```

Comment:
You can hide this by using the [GetParametersByPath paginator](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ssm/paginator/GetParametersByPath.html)

```python
paginator = ssm_client.get_paginator('get_parameters_by_path')
page_iterator = paginator.paginate(Path=path, Recursive=True, WithDecryption=True)
params = sum((page['Contents'] for page in page_iterator), [])
```
If you haven't tinkered with paginators before, give [the boto3 docs on them](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/paginators.html) a quick read.

*The usage of  `sum` might be too much, but you get the idea.*